napari/_vispy/canvas.py

```diff
@@ -1,63 +1,193 @@
     max_texture_sizes : Tuple[int, int]
         The max textures sizes as a (2d, 3d) tuple.
-
+    viewer : napari.components.ViewerModel
+        Napari viewer containing the rendered scene, layers, and controls.
+    view : vispy.scene.widgets.viewbox.ViewBox
+        Rectangular widget in which a subscene is rendered.
+    vispy_camera : napari._vispy.VispyCamera
+        The camera class which contains both the 2d and 3d camera used to describe the perspective by which a
+        scene is viewed and interacted with.
+    _cursors : QtCursorVisual
+        A QtCursorVisual enum with as names the names of particular cursor styles and as value either a staticmethod
+        creating a bitmap or a Qt.CursorShape enum value corresponding to the particular cursor name. This enum only
+        contains cursors supported by Napari in Vispy.
+    _key_map_handler : napari.utils.key_bindings.KeymapHandler
+        KeymapHandler handling the calling functionality when keys are pressed that have a callback function mapped.
+    _last_theme_color : Optional[npt.NDArray[np.float]]
+        Theme color represented as numpy ndarray of shape (4,) before theme change
+        was applied.
+    _overlay_to_visual : dict(napari.components.overlays, napari._vispy.overlays)
+        A mapping of the napari overlays that are part of the viewer and their corresponding Vispy counterparts.
+    _scene_canvas : napari._vispy.canvas.NapariSceneCanvas
+        SceneCanvas which automatically draws the contents of a scene. It is ultimately a VispySceneCanvas, but allows
+        for ignoring mousewheel events with modifiers.
     """
 
     _instances = WeakSet()
 
-    def __init__(self, *args, **kwargs) -> None:
+    def __init__(
+        self,
+        viewer: ViewerModel,
+        key_map_handler: KeymapHandler,
+        *args,
+        **kwargs,
+    ) -> None:
         # Since the base class is frozen we must create this attribute
         # before calling super().__init__().
         self.max_texture_sizes = None
         self._last_theme_color = None
         self._background_color_override = None
-        super().__init__(*args, **kwargs)
+        self.viewer = viewer
+        self._scene_canvas = NapariSceneCanvas(
+            *args, keys=None, vsync=True, **kwargs
+        )
+        self.view = self.central_widget.add_view(border_width=0)
+        self.vispy_camera = VispyCamera(
+            self.view, self.viewer.camera, self.viewer.dims
+        )
+        self.layer_to_visual = {}
+        self._overlay_to_visual = {}
+        self._key_map_handler = key_map_handler
         self._instances.add(self)
 
         # Call get_max_texture_sizes() here so that we query OpenGL right
         # now while we know a Canvas exists. Later calls to
         # get_max_texture_sizes() will return the same results because it's
         # using an lru_cache.
         self.max_texture_sizes = get_max_texture_sizes()
+        self._cursors = QtCursorVisual
```

Comment:
Why does this need to be saved as an instance attribute? Seems like you could just use `QtCursorVisual` itself throughout the class.

Reply:
Agreed, does not have to be saved I think. Changed with [c9e87fa](https://github.com/napari/napari/pull/5432/commits/c9e87fa6c508c0f8c64b0b91dabd1cbb6328b019)